streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -310,7 +313,16 @@ public StreamThread(TopologyBuilder builder,
         log.info("{} Creating producer client", logPrefix);
         this.producer = clientSupplier.getProducer(config.getProducerConfigs(threadClientId));
         log.info("{} Creating consumer client", logPrefix);
-        this.consumer = clientSupplier.getConsumer(config.getConsumerConfigs(this, applicationId, threadClientId));
+
+        Map<String, Object> consumerConfigs = config.getConsumerConfigs(this, applicationId, threadClientId);
+
+        if (!builder.latestResetTopicsPattern().pattern().equals("") || !builder.earliestResetTopicsPattern().pattern().equals("")) {
```

Comment:
I could leave out the updates to the configs, but if the global config is a valid value, we don't have the opportunity to do the specified seeks.   I agree on the use of the  `consumer.committed` method but I don't see how without throwing the `NoOffsetForPartitionException`.